pydicom/dataelem.py

```diff
@@ -848,30 +831,125 @@ def DataElement_from_raw(
                 VR = dictionary_VR(raw.tag)
             except KeyError:
                 pass
+    if VR != raw.VR:
+        raw = raw._replace(VR=VR)
+    return raw
+
+
+def pydicom_raw_elem_convert_value(
+    raw: RawDataElement,
+    encoding: Optional[Union[str, MutableSequence[str]]] = None,
+    **_kwargs,
+) -> RawDataElement:
+    """Attempt to convert the raw value to a native type.
+
+    Parameters
+    ----------
+    raw_data_element : RawDataElement
+        The raw data element with the VR and value to convert.
+    encoding : str or list of str, optional
+        The character encoding of the raw data.
+
+    Returns
+    -------
+    RawDataElement
+    """
+    # XXX buried here to avoid circular import
+    # filereader->Dataset->convert_value->filereader
+    # (for SQ parsing)
+
+    from pydicom.values import convert_value
+    assert raw.VR
```

Comment:
You made a comment about circular imports - could import within functions like was done here to avoid circular.